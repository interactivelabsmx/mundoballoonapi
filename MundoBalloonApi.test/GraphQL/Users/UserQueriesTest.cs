using HotChocolate;
using HotChocolate.Execution;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using MundoBalloonApi.graphql.Users;
using MundoBalloonApi.infrastructure.Data.Models;
using MundoBalloonApi.test.Fixtures;
using Snapshooter.Xunit;
using Xunit;

namespace MundoBalloonApi.test.GraphQL.Users;

public class UserQueriesTest
{
    [Fact]
    public async Task CreateUser()
    {
        // arrange
        var executor = await new ServiceCollection()
            .AddPooledDbContextFactory<MundoBalloonContext>(
                options => options.UseInMemoryDatabase("MundoBalloonContext:Integration:Graphql"))
            .AddGraphQLServer()
            .AddQueryType(d => d.Name("Query"))
            .AddTypeExtension<UserQueries>()
            .BuildRequestExecutorAsync();
        // act
        var fakeData = new FakeData();
        var db = executor.Services.GetService<MundoBalloonContext>();
        await using (db)
        {
            db?.Users.AddRange(fakeData.Users);
            await db?.SaveChangesAsync()!;
        }

        var result = await executor.ExecuteAsync(@"
          userById(userId: 2) {
            ...User
          }
        ");

        // assert
        (await result.ToJsonAsync()).MatchSnapshot();
    }
}