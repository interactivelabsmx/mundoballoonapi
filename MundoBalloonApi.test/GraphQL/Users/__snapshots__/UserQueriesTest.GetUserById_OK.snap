{
  "data": {
    "userById": {
      "id": 999
    }
  }
}
